### PR TITLE
Fix/336

### PR DESCRIPTION
**Pull request by xross**
Addresses #336 
- Fixed some basic bugs in AN000132
- Fixed an issue where device traps when an EP is used by host but not enabled in XUD
- Added a test for the above (test_unused_ep)
- Rationalised some var name and added some comments while I was at it
- Updated changelog and regressed version number in module_build_info - was updated to make changelog checker pass

Changelog check test will fail until https://github.com/xmos/xmos_jenkins_shared_library/pull/266 is merged.

Suggest review with "hide whitespace" since I resolved some indent inconsistencies

### CODE REVIEW

File: lib_xud/src/core/XUD_Main.xc [hunk 1]
```diff
@@ -59,11 +59,13 @@ in port rx_rdy                 = PORT_USB_RX_READY;
...
-// We use a single array instrad of two here and append epAddr_setup on the end to save some instructions in the Setup 
+// We use a single array instrad of two here and append epAddr_Ready_setup on the end to save some instructions in the Setup 
 // token handling code. i.e. what we really want is the following, but's less efficient. 
-// unsigned epAddr[USB_MAN_NUM_EP]
-// unsigned epAddr[USB_MAX_NUM_EP_OUT]
-unsigned epAddr[USB_MAX_NUM_EP + USB_MAX_NUM_EP_OUT];
+// unsigned epAddr_Ready[USB_MAN_NUM_EP]
+// unsigned epAddr_Ready[USB_MAX_NUM_EP_OUT]
+unsigned epAddr[USB_MAX_NUM_EP];                        // Used to store the addr of each EP in ep_info array
+unsigned epAddr_Ready[USB_MAX_NUM_EP + USB_MAX_NUM_EP_OUT];    // Used by the EP to mark itself as ready, essentially same as epAddr_Ready with 0 entries.
```
Comment (danielpieczko) [inline lib_xud/src/core/XUD_Main.xc:67]: I don't understand this comment: this is `epAddr_Ready` and it says it's the "same as epAddr_Ready..."?

**Review by danielpieczko — APPROVED** [review]

File: lib_xud/src/core/XUD_Main.xc (hunk 1 above)
Comment (xross) [reply lib_xud/src/core/XUD_Main.xc:67]: Good spot, should just be `epAddr`

**Comment by xross** [comment]
The failing tests pass locally, investigating.

**Comment by xross** [comment]
> The failing tests pass locally, investigating.

They are now passing, concerning.

**Pull request merged**